packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -0,0 +1,159 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/analytics_client.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/key_value_store.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/sdk/pinpoint.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_secure_storage_dart/amplify_secure_storage_dart.dart';
+import 'package:aws_signature_v4/aws_signature_v4.dart';
+
+/// {@template amplify_analytics_pinpoint_dart.amplify_analytics_pinpoint_dart}
+/// The AWS Cognito Dart implementation of the Amplify Analytics category.
+///
+/// - Validates and parses inputs
+/// - Receives and provides external Flutter Provider implementations
+/// - Delegates work to the [AnalyticsClient]
+/// {@endtemplate}
+class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
+  /// {@macro amplify_analytics_pinpoint_dart.amplify_analytics_pinpoint_dart}
+  AmplifyAnalyticsPinpointDart({
+    SecureStorageInterface? keyValueStorage,
+    CachedEventsPathProvider? pathProvider,
+    AppLifecycleProvider? appLifecycleProvider,
+    DeviceContextInfoProvider? deviceContextInfoProvider,
+  })  : _keyValueSecureStorage = keyValueStorage,
+        _pathProvider = pathProvider,
+        _appLifecycleProvider = appLifecycleProvider,
+        _deviceContextInfoProvider = deviceContextInfoProvider;
+
+  AnalyticsClient? __analyticsClient;
+  AnalyticsClient get _analyticsClient {
+    if (__analyticsClient == null) {
+      throw const AnalyticsException(
+        'Analytics not configured',
+        recoverySuggestion:
+            'Please make sure to call: await Amplify.configure(amplifyconfig)',
+      );
+    }
+    return __analyticsClient!;
+  }
+
+  final SecureStorageInterface? _keyValueSecureStorage;
+
+  /// External Flutter Provider implementations
+  final CachedEventsPathProvider? _pathProvider;
+  final AppLifecycleProvider? _appLifecycleProvider;
+  final DeviceContextInfoProvider? _deviceContextInfoProvider;
```

Comment:
Yeah, null does not interfere with basic analytics behavior. 

Here's why: 

**pathProvider** - not required for web, but otherwise required
**appLifecycleProvider** - if null, sessions are not automatically tracked 
**deviceContextInfoProvider** - if null, no device context info is sent when Pinpoint Endpoint is registered. 
**keyValueStore** - if null, defaults to AmplifySecureStorageWorker